### PR TITLE
GH pane QoL fixes

### LINKED ISSUE #2903: GH pane QoL fixes

### PR DESCRIPTION
**Pull request by ruggi**
Fixes #2903

This PR addresses a bunch of small-ish issues with the GH pane. To make everything easier to review, one commit per fix.

TL;DR:
- don't show remote changes/local changes when the branch is not selected
- allow clearing the repo and the branch
- don't allow collapsing repo/branch blocks if they're not ready
- normalize GH data when something changes
- don't show the `main` default fallback while the branches are loading
- fix icon colors in the load from branch/push to branch buttons of the Contents block
- make all the buttons and CTAs of the pane less susceptible to github operations and mark them as disabled only if they really need to
- make the alternative button for commishing more prominent

### CODE REVIEW

File: editor/src/components/editor/editor-component.tsx [hunk 1]
```diff
@@ -422,6 +427,22 @@ const useGithubData = (): void => {
     'Github data',
   )
 
+  // normalize github data
+  React.useEffect(() => {
+    dispatch([
+      EditorActions.updateGithubSettings({
```
Comment (seanparsons) [inline editor/src/components/editor/editor-component.tsx:433]: Can this normalisation be done in the actions? As currently this means a change to the dependency variables will cause this effect to fire and then dispatch a new action potentially triggering another re-render.

Comment (ruggi) [reply editor/src/components/editor/editor-component.tsx:433]: good point yeah

**Review by bkrmendy — APPROVED** [review]

**Review by seanparsons — APPROVED** [review]

File: editor/src/components/navigator/left-pane/github-pane/index.tsx [hunk 2]
```diff
@@ -135,72 +143,87 @@ const RepositoryBlock = () => {
...
+    (store) => ({
+      currentBranch: store.editor.githubSettings.branchName,
+      dispatch: store.dispatch,
+      githubOperations: store.editor.githubOperations,
+      targetRepository: store.editor.githubSettings.targetRepository,
+      branchesForRepository: store.editor.githubData.branches,
+      repositoryData:
+        store.editor.githubData.publicRepositories.find(
+          (r) => r.fullName === githubRepoFullName(store.editor.githubSettings.targetRepository),
+        ) || null,
```
[the remark this answers is not in the document]
Comment (ruggi) [reply editor/src/components/navigator/left-pane/github-pane/index.tsx:163]: urrrrrrgh, I can't even

File: editor/src/components/navigator/left-pane/github-pane/github-file-changes-list.tsx [hunk 3]
```diff
@@ -188,6 +188,18 @@ export const GithubFileChangesList: React.FC<{
...
   )
 
+  const githubOperations = useEditorState(
+    (store) => store.editor.githubOperations,
+    'Github operations',
+  )
+
+  const disableButtons = React.useMemo(() => {
+    return (
+      githubOperations.some((op) => op.name === 'loadBranch') ||
```
Comment (Rheeseyb) [inline editor/src/components/navigator/left-pane/github-pane/github-file-changes-list.tsx:198]: `isGithubLoadingAnyBranch`

File: editor/src/components/navigator/left-pane/github-pane/index.tsx [hunk 4]
```diff
@@ -675,31 +675,34 @@ const LocalChangesBlock = () => {
...
+                <>
+                  <PushIcon />
+                  Commit and Push
+                </>
+              )}
+            </Button>
+            <div style={{ textAlign: 'center' }}>or</div>
+            <Button onClick={togglePushToNewBranch} spotlight highlight>
+              {when(pushToNewBranch, <span>Push To Current Branch</span>)}
+              {unless(pushToNewBranch, <span>Push To New Branch</span>)}
```
Comment (Rheeseyb) [inline editor/src/components/navigator/left-pane/github-pane/index.tsx:703]: These two conditions appear to be reversed

File: editor/src/components/navigator/left-pane/github-pane/github-file-changes-list.tsx (hunk 3 above)
Comment (ruggi) [reply editor/src/components/navigator/left-pane/github-pane/github-file-changes-list.tsx:198]: good spot

Comment (ruggi) [reply editor/src/components/navigator/left-pane/github-pane/github-file-changes-list.tsx]: https://github.com/concrete-utopia/utopia/pull/2904/commits/9190b1588669b0d2acff2343c660de5b4690d3ee

File: editor/src/components/editor/editor-component.tsx (hunk 1 above)
Comment (ruggi) [reply editor/src/components/editor/editor-component.tsx]: https://github.com/concrete-utopia/utopia/pull/2904/commits/ec20bdc04400531b53d6b8c9e586673cf6e01aa9

File: editor/src/components/navigator/left-pane/github-pane/index.tsx (hunk 4 above)
Comment (ruggi) [reply editor/src/components/navigator/left-pane/github-pane/index.tsx:703]: yeah it's tricky - but they are correct, they are on the negative logic side. These buttons are displayed to perform the _opposite_ action

File: editor/src/components/navigator/left-pane/github-pane/index.tsx [hunk 5]
```diff
@@ -358,16 +407,16 @@ const BranchBlock = () => {
   )
 }
 
-const PullIcon = () => {
+const PullIcon = ({ fill }: { fill?: string }) => {
```
Comment (Rheeseyb) [inline editor/src/components/navigator/left-pane/github-pane/index.tsx:410]: This change is going to conflict with changes from #2901

Comment (ruggi) [reply editor/src/components/navigator/left-pane/github-pane/index.tsx:410]: huh - alright, let's merge #2901 and resolve here

**Review by balazsbajorics — APPROVED** [review]

**Review by Rheeseyb — APPROVED** [review]

File: editor/src/components/navigator/left-pane/github-pane/index.tsx (hunk 4 above)
Comment (Rheeseyb) [reply editor/src/components/navigator/left-pane/github-pane/index.tsx]: Ahhh I see!

**Pull request merged**